src/server/auth/server/api_server.go

```diff
@@ -2258,16 +2206,16 @@ func (a *apiServer) canonicalizeSubject(ctx context.Context, subject string) (st
 		colonIdx = len(authclient.GitHubPrefix) - 1
 	}
 	prefix := subject[:colonIdx]
-	a.configMu.Lock()
-	defer a.configMu.Unlock()
 
 	// check prefix against config cache
-	if a.configCache != nil {
-		if prefix == a.configCache.IDP.Name {
-			return subject, nil
-		}
-		if prefix == path.Join("group", a.configCache.IDP.Name) {
-			return subject, nil // TODO(msteffen): check if this IdP supports groups
+	if config := a.getCacheConfig(); config != nil {
+		for _, idp := range config.IDPs {
```

Comment:
From our discussion, we are going to continue to canonicalize the subject based only on the IDP's name and not on anything that is in the various *Options fields, correct?

Reply:
I'm not 100% sure I undrestand the question, but I think that's right (this is also what we currently, do, if I understand the question).

To be clear, and maybe a little pendatic
1) I think the plan for now is that post-change, an IDP's `name` is `github` if and only if it has a `GitHubOptions`. It's just a matter of whether there's an IDP with `GitHubOptions`), or equivalently, an IDP whose `name` is `github`.
2) a SAML IDP can give you two subject types if it supports groups:
- `<idp.name>:<subject>` (e.g. `okta:matthew`)
- `group/<idp.name>:<subject>` (e.g. `group/okta:engineering`).
but we don't currently check the IDP to make sure `saml.group_attribute` is set before allowing `group/<idp_name>`-type subjects through `canonicalizeSubject`. Perhaps we should but I think it's fine to keep that outside the scope of this change.